builtin/logical/pki/path_fetch.go

```diff
@@ -324,6 +339,19 @@ reply:
 		return
 	case response.IsError():
 		return response, nil
+	case httpStatusCode != 0:
+		response = &logical.Response{
+			Data: map[string]interface{}{
+				logical.HTTPContentType: contentType,
+				logical.HTTPRawBody:     certificate,
```

Comment:
I think we want to unset this if the status code is 304? Otherwise I believe clients will see the whole response, which I don't think is correct? We should probably test and validate this against say, curl or something. 

I think https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/304#compatibility_notes and https://www.webfx.com/web-development/glossary/http-status-codes/what-is-a-304-status-code/ backs up not sending a body in this case.

Reply:
I used curl during the development to test the endpoint, from what I understand, with the current implementation and since these values are still default when set, the body is empty and content-type isn't returned either.
I have now tried to unset both values, and if content-type isn't set, the response is 500. (Body seems to be optional).

Reply:
Ah interesting. That's fine then :-) I hadn't realized the body wouldn't be sent if we had specified it in the response object. Kinda weird tbh.